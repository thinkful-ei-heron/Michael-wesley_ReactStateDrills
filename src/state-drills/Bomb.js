import React, { Component } from 'react'

export class Bomb extends Component {
  constructor(props) {
    super(props)
    this.state = {count: 0};
  }


  componentDidMount() {
    const second = 1000; //ms

    // let interval = setInterval(callback, timeInMs)
    let interval = window.setInterval(
      () => {
      let newCount = this.state.count + 1;

      this.setState({
        count: newCount
      })

      console.log(this.state.count);
    },second)

    this.setState({
      interval
    })
  }

  componentWillUnmount() {
    clearInterval(this.state.interval);
    this.setState({
      interval: null
    })
  }


  render() {
    let str;
    if (this.state.count % 2){ // 0 is falsy, 1 is truthy
      str = 'tock'
    } else {
      str = 'tick'
    }
    if (this.state.count >= 8) {
      str = 'BOOM!!!!';
      clearInterval(this.state.interval)
      this.setState({
        interval: null
      })
    }
    return (
      <div>
        <p>{str}</p>
      </div>
    )
  }
}

export default Bomb
