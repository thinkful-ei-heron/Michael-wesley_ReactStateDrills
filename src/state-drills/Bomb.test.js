import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Bomb from './Bomb';



describe('Bomb test', () => {
  it('renders without crashing', () => {
    const result = document.createElement('div');
    ReactDOM.render(<Bomb/>, result);
    ReactDOM.unmountComponentAtNode(result);
  })
  it('renders as expected', ()=>{
    expect(renderer.create(<Bomb />)
    .toJSON())
    .toMatchSnapshot()
  })

})
