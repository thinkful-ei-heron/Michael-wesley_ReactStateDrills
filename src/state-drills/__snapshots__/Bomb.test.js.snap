// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bomb test renders as expected 1`] = `
<div>
  <p>
    tick
  </p>
</div>
`;
